### DISCUSSION TITLE
At x509_ Build in vfy. c_ In the chain function, ok=num>max_ Depth? 0: get1_ Trusted_ Issuer (&issuer, ctx, current); The problem of determining the return value

### DISCUSSION THREAD

**Discussion by lurio-liu** [openssl/openssl#22435] [Q&A] [answered]
static int build_chain(X509_STORE_CTX *ctx)
           {
                    ......
                    ok = num > max_depth ? 0 : get1_trusted_issuer(&issuer, ctx, curr);

                    if (ok < 0) {
                        trust = -1;
                        ctx->error = X509_V_ERR_STORE_LOOKUP;
                        break;
                    }
                    if (ok > 0) {
                        ......
                    }
           }
           The return value of ok comes from get1_trusted_issuer,  The return value of the get1_trusted_issuer function comes from X509_STORE_CTX_get1_issuer.
           The return value of the X509_STORE_CTX_get1_issuer is described as 
            * Return values are:
           *  1 lookup successful.
           *  0 certificate not found.
           * -1 some other error.
           */
          But the return value 0 was not processed in the build_chain function.
          "certificate not found" was not "X509_V_ERR_STORE_LOOKUP"?

**Comment by t8m** [comment, marked as answer]
The ok == 0 means that the trusted issuer was not found yet and the search continues by other means.

This is not a bug.
